Software/Development/SemesterProject/App/Include/digitalInput.h

```diff
@@ -13,8 +13,8 @@
 //
 typedef struct DigitalInputTag
 {
-    int pin;
-    int state;
+    unsigned char pin:6;
+    unsigned char state:1;
 } DigitalInput;
```

Comment:
Why doing this?

Reply:
I saw that you didn't do it this way in the final version, gut.

Reply:
if you mean the colon with the number, it's like that in the final implementation. It basically is telling the system that although the type is a char, we are only going to use 6 or 1 bit. 

This was suggested by Lajos this week. He said it was good for omtimization, however I find it more useful for adding safety to the system, as values different to the appropriate one will lead to a value of 0. However maybe this 0 should be treated somehow, might be something to look at in errorManager.

Reply:
Ok, i meant more why using char, but thanks for the explanation.

Reply:
I used char, as suggested by Lajos to save memory. However using char for small integer sometimes (very specific times usually when talking about data representation (casting, printf,...))leads to the compiler not handling the integer as a number but as a character.